pkgs/development/compilers/ghc/8.0.1.nix

```diff
@@ -57,6 +74,15 @@ stdenv.mkDerivation rec {
   stripDebugFlags = [ "-S" ] ++ stdenv.lib.optional (!stdenv.isDarwin) "--keep-file-symbols";
 
   postInstall = ''
+    # ghci uses mmap with rwx protection at it implements dynamic
+    # linking on its own. See:
+    # - https://bugs.gentoo.org/show_bug.cgi?id=299709
+    # - https://ghc.haskell.org/trac/ghc/ticket/4244
+    # Therefore, we have to pax-mark the resulting binary.
+    # Haddock also seems to run with ghci, so mark it as well.
+    paxmark m $out/lib/${name}/bin/{ghc,haddock}
```

Comment:
I believe the `ghci` reference is indeed intentional, it matches the discussions on the referenced gentoo bug and haskell ticket, reproduced below for convenience:

* https://bugs.gentoo.org/show_bug.cgi?id=299709
* https://ghc.haskell.org/trac/ghc/ticket/4244

See also: https://ghc.haskell.org/trac/ghc/ticket/12657

I'm not sure where the commentary belongs, but it does seem unfortunate to have multiple copies of it.  I'll poke at this some more, thanks for the review.